addons/mail/static/src/models/message/message.js

```diff
@@ -126,7 +126,7 @@ function factory(dependencies) {
             if ('starred_partner_ids' in data) {
                 data2.isStarred = data.starred_partner_ids.includes(this.env.messaging.currentPartner.id);
             }
-            if ('subject' in data) {
+            if ('subject' in data && data.subject !== this.env._t(`Re: ${data.record_name}`)) {
```

Comment:
1. don't do the fix in convertData. The data is "real" no matter what. This should be a "display" fix.
2. don't translate it, "Re:" is standard in email stuff and people can send emails in whatever language they want. If there are other prefix that we need to handle, we should have a list of them.

Reply:
For point 1. I suggest a computed field on message model js, (name suggestion: `cleanedSubject`)